### PR TITLE
HTTP status codes as described in documentation.

### PR DESCRIPTION
**Pull request by HussainMir**
https://docs.microsoft.com/en-us/rest/api/documentdb/http-status-codes-for-documentdb

### CODE REVIEW

**Comment by kirankumarkolli** [comment]
@srinathnarayanan  will take a look.

**Comment by srinathnarayanan** [comment]
Hi @HussainMir . Are these additions of Http status codes to the python sdk just to make it consistent with other sdk s, or have you found any places in the code base where we use the raw error code value which can be replaced by these constants? Thanks!

**Comment by HussainMir** [comment]
Hi, is there any testing needs to be done from my side? Wondering if this pull request can be improved somehow, or modified? 

Thanks.

File: pydocumentdb/execution_context/execution_dispatcher.py [hunk 1]
```diff
@@ -90,7 +91,7 @@ def fetch_next_block(self):
         return self._execution_context.fetch_next_block()        
         
     def _is_partitioned_execution_info(self, e):    
-        return e.status_code == 400 and e.sub_status == 1004
```
Comment (srinathnarayanan) [inline pydocumentdb/execution_context/execution_dispatcher.py]: Can you add SubStatusCodes as well, in the constants file? And change references to 1004 and other sub status codes to constants.

File: pydocumentdb/http_constants.py [hunk 2]
```diff
@@ -266,3 +266,25 @@ class HttpContextProperties:
     """Constants of http context properties.
     """
     SubscriptionId = 'SubscriptionId'
+
+
+class HttpStatusCodes:
```
Comment (srinathnarayanan) [inline pydocumentdb/http_constants.py]: Rename to StatusCodes and SubStatusCodes to be consistent with .Net SDK

File: test/crud_tests.py [hunk 3]
```diff
@@ -140,7 +140,7 @@ def _test_database_crud(self, is_name_based):
         # delete database.
         client.DeleteDatabase(self.GetDatabaseLink(created_db, is_name_based))
         # read database after deletion
-        self.__AssertHTTPFailureWithStatus(404,
+        self.__AssertHTTPFailureWithStatus(http_constants.StatusCodes.NOT_FOUND,
```
Comment (srinathnarayanan) [inline test/crud_tests.py]: can you "import http_constants.StatusCodes as StatusCodes" and replace references to http_constants.StatusCodes with just StatusCodes ? here and everywhere else

Comment (srinathnarayanan) [reply test/crud_tests.py]: Similarly for SubStatusCodes as well

**Pull request merged**